### PR TITLE
updated terraform files, added modued, iam role and policy, node expo…

### PR DESCRIPTION
**Pull request by PavloTarnovetskyi**
…rter

What is this change?
edited terraform files, added modules, iam role policy. In Dockerfile added installation of boto3 module and in requirements added node exporter module.

**Why is this important?**
It allows ec2 access to ssm parameters and takes token from.

**How this has been tested?**

manually

**How can this be rolled back?**

Rollback can be performed by reverting these changes in source control and redeploying via standard deployment mechanisms. This change is safe to roll back.

**Are there any security risks introduced by this change?**

### CODE REVIEW

File: app&dockerfile/Dockerfile [hunk 1]
```diff
@@ -4,7 +4,8 @@ WORKDIR /app
 
 COPY . /app
 
-RUN pip3 --no-cache-dir install -r requirements.txt                                                                            
+RUN pip3 --no-cache-dir install -r requirements.txt \
```
Comment (nos373737) [inline app&dockerfile/Dockerfile]: you can add boto3 to requeirements.txt here

Comment (nos373737) [inline app&dockerfile/Dockerfile]:
```suggestion
RUN pip3 --no-cache-dir install -r requirements.txt
```

File: app&dockerfile/Dockerfile [hunk 2]
```diff
@@ -4,7 +4,8 @@ WORKDIR /app
...
+    && pip install boto3
```
Comment (nos373737) [inline app&dockerfile/Dockerfile]:
```suggestion
```

File: app&dockerfile/requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
 Flask==2.0.3
-Flask-RESTful==0.3.9
+requests==2.27.1
 Werkzeug==2.0.3
-requests==2.27.1
\ No newline at end of file
+prometheus-flask-exporter
```
Comment (nos373737) [inline app&dockerfile/requirements.txt]:
```suggestion
prometheus-flask-exporter
boto3
```

File: terraform/ec2/ec2.tf [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (nos373737) [inline terraform/ec2/ec2.tf]:
```suggestion
```

Comment (nos373737) [reply terraform/ec2/ec2.tf]: and empty lines below can also be deleted

File: terraform/ssh/ssh.tf [hunk 5]
```diff
@@ -0,0 +1,12 @@
+variable "ssh" {
+  type = string
+}
+
+
+resource "aws_key_pair" "ssh-key" {
+  key_name   = "aws-key"
+  public_key = file("/Users/pavlo/.ssh/aws-key.pub")
```
Comment (nos373737) [inline terraform/ssh/ssh.tf:8]:
```suggestion
  public_key = file("~$HOME/.ssh/aws-key.pub")
```

Comment (nos373737) [reply terraform/ssh/ssh.tf:8]: Or something similar, you can test it locally

**Comment by nos373737** [comment]
After all suggestion above you can merge it.

**Comment by nos373737** [comment]
Please attach `terraform plan` output via github gist in the comment, then we will be able to review it.

**Comment by nos373737** [comment]
TF plan https://gist.github.com/nos373737/54c099c9b9e10b771869dbad67a3019f

**Comment by PavloTarnovetskyi** [comment]
https://gist.github.com/PavloTarnovetskyi/445840a57eebf2a14228944274f7d3a8

**Comment by PavloTarnovetskyi** [comment]
TF_plan https://gist.github.com/PavloTarnovetskyi/445840a57eebf2a14228944274f7d3a8

File: terraform/ssh/ssh.tf (hunk 5 above)
Comment (nos373737) [reply terraform/ssh/ssh.tf:8]: false alarm here

**Review by nos373737 — APPROVED** [review]
lgtm

**Review by antoniakstas — COMMENTED** [review]
lgtm

**Pull request merged**